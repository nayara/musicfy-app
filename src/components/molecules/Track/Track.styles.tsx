import { styled } from "styled-components";

const Title = styled.h1`
  color: ${({ theme }) => theme.colors.neutral.white};
  font-size: 3vh;
`;

const Wrapper = styled.div`
  display: flex;
  overflow-x: scroll;
  overflow-y: hidden;
`;

const Track = styled.div`
  display: flex;
  flex-direction: column;
`;

export { Title, Wrapper, Track };
