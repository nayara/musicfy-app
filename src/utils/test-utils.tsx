import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
