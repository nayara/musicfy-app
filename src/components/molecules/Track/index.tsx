import { FC } from "react";
import { TContentItem } from "../../../api/queries/use-get-home";
import Card from "../../../components/atoms/Card";
import * as Style from "./Track.styles";

type TTrack = {
  title: string;
  contentsItems: TContentItem[];
};

const handleClick = () => {
  alert("To be implemented");
};

const Track: FC<TTrack> = ({ title, contentsItems }) => {
  return (
    <>
      <Style.Title>{title}</Style.Title>
      <Style.Wrapper>
        {contentsItems.map((item: TContentItem, index) => (
          <Style.Track key={index}>
            <Card
              imgSrc={item.images[0][0].url}
              altText={`Playlist ${item.name} cover`}
              title={item.name}
              onClick={handleClick}
              key={`card-${index}`}
            />
          </Style.Track>
        ))}
      </Style.Wrapper>
    </>
  );
};

export default Track;
