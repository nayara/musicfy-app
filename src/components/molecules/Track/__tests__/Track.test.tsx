import { render, screen } from "../../../../utils/test-utils";
import Track from "..";
import userEvent from "@testing-library/user-event";

const mockedItems = [
  {
    type: "playlist",
    id: "some-id",
    name: "Happy Hits!",
    shareUrl: "https://open.spotify.com/playlist/37i9dQZF1DXdPec7aLTmlC",
    description: "Hits to boost your mood and fill you with happiness!",
    owner: {
      type: "user",
      name: "Spotify",
    },
    images: [
      [
        {
          url: "https://i.scdn.co/image/ab67706f00000002b55b6074da1d43715fc16d6d",
          width: null,
          height: null,
        },
      ],
    ],
  },
];

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useNavigate: () => mockNavigate,
}));

describe("<Track />", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Track title="Some awsome track" contentsItems={mockedItems} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should redirect to playlist page when playlist card is clicked", async () => {
    render(<Track title="Some awsome track" contentsItems={mockedItems} />);

    const cardButton = screen.getByRole("button", {
      name: /Playlist Happy Hits!/i,
    });

    await userEvent.click(cardButton);

    expect(mockNavigate).toHaveBeenCalledWith("/playlist/some-id");
  });
});
