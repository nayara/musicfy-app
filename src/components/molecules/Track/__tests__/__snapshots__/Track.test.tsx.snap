// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> should render correctly 1`] = `
.c3 {
  display: flex;
  align-items: center;
  border-radius: 0.5rem;
  padding: 1rem;
  flex-direction: column;
  transition: transform 0.45s;
}

.c3:hover {
  background-color: rgba(0, 0, 0, 0.2);
}

.c4 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

.c7 {
  text-align: left;
  color: #EEF1EF;
  font-size: 1.3rem;
  align-self: flex-start;
  margin: 0;
  padding: 0.3rem 0;
}

.c8 {
  text-align: left;
  color: #A9B4C2;
  font-size: 1rem;
  align-self: flex-start;
  margin: 0;
}

.c5 {
  border-radius: 0.5rem;
  height: 10rem;
}

.c6 {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 0.9rem 0;
}

.c0 {
  color: #EEF1EF;
  font-size: 1.5rem;
}

.c1 {
  display: flex;
}

.c2 {
  display: flex;
  flex-direction: column;
}

<div>
  <h1
    class="c0"
  >
    Some awsome track
  </h1>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <button
          class="c4"
        >
          <img
            alt="Playlist Happy Hits! cover"
            class="c5"
            src="https://i.scdn.co/image/ab67706f00000002b55b6074da1d43715fc16d6d"
          />
          <div
            class="c6"
          >
            <h2
              class="c7"
            >
              Happy Hits!
            </h2>
            <h3
              class="c8"
            >
              some subtitle
            </h3>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
